pyscf/pbc/cc/kccsd_rhf.py

```diff
@@ -728,25 +729,29 @@ def __init__(self, cc, mo_coeff=None, method='incore'):
 
         mo_coeff = self.mo_coeff = padded_mo_coeff(cc, mo_coeff)
 
-        # Re-make our fock MO matrix elements from density and fock AO
-        dm = cc._scf.make_rdm1(cc.mo_coeff, cc.mo_occ)
-        with lib.temporary_env(cc._scf, exxdiv=None):
-            # _scf.exxdiv affects eris.fock. HF exchange correction should be
-            # excluded from the Fock matrix.
-            fockao = cc._scf.get_hcore() + cc._scf.get_veff(cell, dm)
-        self.fock = np.asarray([reduce(np.dot, (mo.T.conj(), fockao[k], mo))
-                                for k, mo in enumerate(mo_coeff)])
-
-        self.mo_energy = [self.fock[k].diagonal().real for k in range(nkpts)]
-        # Add HFX correction in the self.mo_energy to improve convergence in
-        # CCSD iteration. It is useful for the 2D systems since their occupied and
-        # the virtual orbital energies may overlap which may lead to numerical
-        # issue in the CCSD iterations.
-        # FIXME: Whether to add this correction for other exxdiv treatments?
-        # Without the correction, MP2 energy may be largely off the correct value.
-        madelung = tools.madelung(cell, kpts)
-        self.mo_energy = [_adjust_occ(mo_e, nocc, -madelung)
-                          for k, mo_e in enumerate(self.mo_energy)]
+        if cc.keep_exxdiv:
+            self.fock = np.asarray([np.diag(mo_e) for k, mo_e in enumerate(cc._scf.mo_energy)], dtype=dtype)
+            self.mo_energy = [self.fock[k].diagonal().real for k in range(nkpts)]
```

Comment:
There is no guarantee that orbitals diagonalize the Fock matrix. I think it's better to construct the Fock matrix explicitly than using the orbital energies as the diagonal part.

Reply:
We wanted to test whether keeping the exxdiv effect in electron correlation will improve the k-point convergence. Anyway, `keep_exxdiv = False` by default and I have hidden it from the constructors (and added a warning comment to avoid user modification). Or, we can simply remove the changes here and keep them in my local branch.

Reply:
This comment is about the way to construct self.fock and self.mo_energy. Computing fock matrix with the code
```
self.fock = np.asarray([np.diag(mo_e) for k, mo_e in enumerate(cc._scf.mo_energy)], dtype=dtype)
```
requires canonical HF orbitals. I think it's better to compute Fock from definition as the old code did
```
fockao = cc._scf.get_hcore() + cc._scf.get_veff(cell, dm)
self.fock = np.asarray([reduce(np.dot, (mo.T.conj(), fockao[k], mo))
                                for k, mo in enumerate(mo_coeff)])
```
without the temporary environment statement
```
with lib.temporary_env(cc._scf, exxdiv=None):
```

I agree it's useful to add the flag  keep_exxdiv

Reply:
Now in both cases, Fock is computed from density matrix and Fock AO. `keep_exxdiv` is kept. See commit 2da05c3.